Cql/CodeGeneration.NET/CSharpSourceCodeWriterCallbacks.cs

```diff
@@ -23,12 +34,14 @@ internal class CSharpSourceCodeWriterCallbacks
     /// <param name="libraryNameToClassName">Generating a class name for the library name.</param>
     /// <param name="shouldWriteLibrary">A function that determines whether the given library should be generated or not; default is <see langword="null" />.  When <see langword="null" />, all libraries are written.</param>
     /// <param name="onAfterStep">For handling a stream directly after the code was generated.</param>
+    /// <param name="shouldThrowException">Whether exception should be thrown.</param>
     public CSharpSourceCodeWriterCallbacks(
         Func<string, string?>? libraryNameToClassName = null,
         Predicate<string>? shouldWriteLibrary = null,
-        Action<CSharpSourceCodeStep>? onAfterStep = null)
+        Action<CSharpSourceCodeStep>? onAfterStep = null,
+        Func<ExceptionContext, bool>? shouldThrowException = null)
```

Comment:
Are we sure it's clear from the param name that this callback is expected to have side-effects (e.g. use this to write to the log?),.